worker/backend/backend.go

```diff
@@ -106,14 +113,75 @@ func (b *Backend) Create(gdnSpec garden.ContainerSpec) (container garden.Contain
 //
 // Errors:
 // * TODO.
-func (b *Backend) Destroy(handle string) (err error) { return }
+func (b *Backend) Destroy(handle string) error {
+	ctx := namespaces.WithNamespace(context.Background(), b.namespace)
+	const maxTaskKillWaitTime = 10 * time.Second
+
+	if handle == "" {
+		return InputValidationError{}
+	}
+
+	container, err := b.client.GetContainer(ctx, handle)
+	if err != nil {
+		return err
+	}
+
+	task, err := container.Task(ctx, nil)
+	if err != nil {
+		if !errdefs.IsNotFound(err) {
+			return err
+		}
+
+		return b.client.Destroy(ctx, handle)
+	}
+
+	timeDelimitedContext, _ := context.WithTimeout(ctx, maxTaskKillWaitTime)
+	err = killTasks(timeDelimitedContext, task)
+	if err != nil {
+		return err
+	}
+
+	return b.client.Destroy(ctx, handle)
+}
+
+// killTasks kills a task on time, gracefully if possible, ungracefully if not.
+//
+func killTasks(ctx context.Context, task containerd.Task) error {
+	exitStatus, err := task.Wait(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = task.Kill(ctx, syscall.SIGTERM)
```

Comment:
Based on the docs, its not clear, is `task.Kill` async like `task.Wait` ?

Reply:
Just saw [this](https://github.com/concourse/concourse/pull/4784#issuecomment-558836584) so this makes sense now :)